File: ClimaTempo.Web/Entidades/Weather.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ClimaTempo.Web.Entidades
{
    public class Weather
    {
        [JsonProperty("date")]
        DateTime date { get; set; }

        [JsonProperty("text")]
        string text { get; set; }
        
        [JsonProperty("temperature")]
        Temperature temperature { get; set; }
        
        [JsonProperty("rain")]
        Rain rain { get; set; }
    }
}

File: backend/src/models/Weather.ts


class Weather {
  id!: string;
  
  period!: {
    begin: Date,
    end: Date
  };
  
  locale_id!: string;

  weather!: [
    {
      date: Date,
      text: string,
      temperature: {
        min: number,
        max: number
      },
      rain: {
        probability: number,
        precipitation: number
      },
    }
  ];
}